packages/identity-sdk/src/viem-sdk.ts

```diff
@@ -0,0 +1,215 @@
+  params: SimulateContractParameters,
+  publicClient: PublicClient,
+  walletClient: WalletClient & WalletActions,
+  onHash?: (hash: `0x${string}`) => void,
+) => {
+  try {
+    const [account] = await walletClient.getAddresses();
+    if (!account) throw new Error("No active wallet address found.");
+
+    const { request } = await publicClient.simulateContract({
+      account,
+      ...params,
+    });
+
+    const hash = await walletClient.writeContract(request);
+    onHash?.(hash);
+
+    const receipt = await waitForTransactionReceipt(publicClient, { hash });
+
+    if (!receipt) {
+      throw new Error(`Transaction receipt not found for hash ${hash}`);
+    }
+
+    if (receipt.status !== "success") {
+      throw new Error(`Transaction failed with status ${receipt.status}`);
+    }
+
+    return receipt;
+  } catch (error: any) {
+    console.error("submitAndWait Error:", error);
+    throw new Error(`Failed to submit transaction: ${error.message}`);
+  }
+};
+
+export const isWhitelisted = async (
+  contract: IdentityContract,
+  accountToCheck: Address,
+): Promise<boolean> => {
+  const validation = addressSchema.safeParse(accountToCheck);
+  if (!validation.success) {
+    throw new Error(
+      `isWhitelisted Error: ${validation.error.issues.map((issue) => issue.message).join(", ")}`,
+    );
+  }
+
+  try {
+    const result = await contract.publicClient.readContract({
+      address: contract.contractAddress,
+      abi: contract.abi,
+      functionName: "isWhitelisted",
+      args: [accountToCheck],
+    });
+
+    if (typeof result !== "boolean") {
+      throw new Error(
+        `isWhitelisted Error: Expected boolean, received ${typeof result}`,
+      );
+    }
+
+    return result;
+  } catch (error: any) {
+    console.error("isWhitelisted Error:", error);
+    throw new Error(`Failed to check whitelist status: ${error.message}`);
+  }
+};
+
+export const getIdentityExpiryData = async (
+  contract: IdentityContract,
+  account: Address,
+): Promise<IdentityExpiryData> => {
+  const validation = addressSchema.safeParse(account);
+  if (!validation.success) {
+    throw new Error(
+      `getIdentityExpiryData Error: ${validation.error.issues.map((issue) => issue.message).join(", ")}`,
+    );
+  }
+
+  try {
+    const [lastAuthenticated, authPeriod] = await Promise.all([
+      contract.publicClient.readContract({
+        address: contract.contractAddress,
+        abi: contract.abi,
+        functionName: "lastAuthenticated",
+        args: [account],
+      }),
+      contract.publicClient.readContract({
+        address: contract.contractAddress,
+        abi: contract.abi,
+        functionName: "authenticationPeriod",
+        args: [],
+      }),
+    ]);
+
+    if (
+      typeof lastAuthenticated !== "bigint" ||
+      typeof authPeriod !== "bigint"
+    ) {
+      throw new Error("Invalid data types returned from contract.");
+    }
+
+    return { lastAuthenticated, authPeriod };
```

Comment:
authPeriod is in days. i would call it authPeriodDays.
also i would add expirationTimestamp variable (lastAuthenticated + authPeriod*DAY_SECONDS)